public/app/plugins/datasource/elasticsearch/configuration/ElasticDetails.tsx

```diff
@@ -25,8 +25,8 @@ export const ElasticDetails = ({ value, onChange }: Props) => {
         <InlineField label="Index name" labelWidth={26}>
           <Input
             id="es_config_indexName"
-            value={value.database || ''}
-            onChange={changeHandler('database', value, onChange)}
+            value={value.jsonData.index || value.database || ''}
```

Comment:
i recommend:

```suggestion
            value={value.jsonData.index ?? value.database || ''}
```


(i have the same recommendation below somewhere, where i have a detailed description of this)